### PR TITLE
Bignum: Remove diamond inheritance from bignum test generation

### PR DESCRIPTION
**Pull request by yanesca**
## Description

The class BaseTarget served two purposes:
    
- track test cases and target files for generation
- provide an abstract base class for individual test groups
    
Splitting these allows decoupling these two and to have further common superclasses across targets.

## Gatekeeper checklist

- [ ] **changelog** not required
- [ ] **backport** not required
- [ ] **tests** not required

### CODE REVIEW

File: scripts/mbedtls_dev/bignum_core.py [hunk 1]
```diff
@@ -16,20 +16,19 @@
...
-from abc import ABCMeta
 from typing import Dict, Iterator, List, Tuple
 
 from . import test_case
 from . import test_data_generation
 from . import bignum_common
 
-class BignumCoreTarget(test_data_generation.BaseTarget, metaclass=ABCMeta):
-    #pylint: disable=abstract-method
+class BignumCoreTarget(test_data_generation.BaseTarget):
```
Comment (gilles-peskine-arm) [inline scripts/mbedtls_dev/bignum_core.py:25]: Having looked superficially at the changes here but not dived into the details, this looks sensible to me.

To the Python experts: what to we lose by not giving certain classes a metaclass anymore?

**Review by wernerlewis — COMMENTED** [review]
These changes are reasonable to me, I think the design looks good and would definitely help resolve the issues with duplication and complexity.

File: scripts/mbedtls_dev/bignum_core.py (hunk 1 above)
Comment (minosgalanakis) [reply scripts/mbedtls_dev/bignum_core.py:25]: In short @yanesca described it accurately. This is more about maintaining the metaclass properties, as such the strong enforcement that the abstraclasses are implemented by the children.

The rule of thumb is that:
* If you want this class to be inheritted and children have to implement the abstractclasses ==> use metaclass=ABCMeta
* If you want this class to be standlone or the implementation of the abstractmethods defined in it, to be suffient for it's children you can remove the `metaclass=ABCMeta`

**Comment by yanesca** [comment]
Superseded by #6632

**Pull request closed without merging**